resolver-dns/src/main/java/io/netty/resolver/dns/BiDnsQueryLifecycleObserver.java

```diff
@@ -0,0 +1,96 @@
+package io.netty.resolver.dns;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsResponseCode;
+import io.netty.util.internal.UnstableApi;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * Combines two {@link DnsQueryLifecycleObserver} into a single {@link DnsQueryLifecycleObserver}.
+ */
+@UnstableApi
+public final class BiDnsQueryLifecycleObserver implements DnsQueryLifecycleObserver {
```

Comment:
Creation of the list (or varargs array) requires allocation ... this is what I'm trying to avoid and suggested if there is more than 2 required in the future a `MultiDnsQueryLifecycleObserver` can be used which accepts a list/varargs.